include/platform-aws.h

```diff
@@ -72,7 +72,15 @@ class PlatformAWS : public Platform {
 		return fields ? fields->func_idx : -EIO;
 	}
 
+	// Determine AWS at runtime
+	bool is_aws() const;
+
 private:
+	// Constants
+	static constexpr int BASE_PRIORITY = 100;
+	static constexpr unsigned short AWS_VENDOR_ID = 0x1D0F;
```

Comment:
This is the Amazon vendor id, not the aws vendor id.  You also should have PCI in there somewhere.